pandas/core/apply.py

```diff
@@ -131,10 +132,13 @@ def get_result(self):
 
         # ufunc
         elif isinstance(self.f, np.ufunc):
-            with np.errstate(all='ignore'):
-                results = self.f(self.values)
-            return self.obj._constructor(data=results, index=self.index,
-                                         columns=self.columns, copy=False)
+            result = self.obj._constructor(index=self.index, copy=False)
+            for col in self.columns:
+                if is_sparse(self.obj.dtypes[col]):
```

Comment:
This is closer, but I don't think it should be special casing sparse. The bug applies to any extension array.

https://github.com/pandas-dev/pandas/pull/23293 is fixing ufuncs on series. So extracting `.values` won't be necessary. I'd recommend waiting for https://github.com/pandas-dev/pandas/pull/23293 and seeing if you can just do `result[col] = self.f(self.obj[col])`.

Reply:
I removed the sparse check in the last commit because it turns out calling `values` on normal columns is also acceptable (should generalize to other array types as well). I do agree that it would look neater if calling `values` wasn't required at all though.